arangod/Cluster/ClusterMethods.cpp

```diff
@@ -3202,6 +3229,17 @@ arangodb::Result hotRestoreCoordinator(VPackSlice const payload, VPackBuilder& r
     return result;
   }
 
+  // Check if the version matches the current version
+  if (!ignoreVersion) {
+    using arangodb::methods::Version;
+    using arangodb::methods::VersionResult;
+    BackupMeta &meta = list.begin()->second;
```

Comment:
This is testing only the first backup found in an unordered list.  It is not testing a specific hot backup is the expected version.  Is there something that already filtered the "list" to only containing the requested backup image?

Reply:
This is implicitly handled by the api of `hotBackupList`: if `payload` contains an `id` field the dbservers will return only a single backup. hence `list.size() == `. Let me add an assertion.